CONTRIBUTING.md

```diff
@@ -341,5 +373,72 @@ version in `pyproject.toml` to:
 And similarly, in `stcal`, change the required `jwst` version to:
 
 	>> jwst @  git+https://github.com/<your_username>/jwst.git@<your_branch>
+
 Let the CI run and ensure it passes, comment this in your PR and make sure the reviewers
 confirm, and then change the versions back before your PR is merged (which will again cause the CI to fail, but that’s OK).
+
+## Code style
+
+We use a pre-commit CI workflow to ensure that the code and docstring style of the `jwst` repository
+remains uniform and conforms to certain standards. We recommend checking these using `pre-commit`
+as described in the [Installing JWST for Development](Step-3-Installing-jwst-for-development)
+section. For additional information about any of these individual style checkers,
+see their documentation linked below.
+Our pre-commit Git hook, also described in the
+[Installing JWST for Development](Step-3-Installing-jwst-for-development) section,
+is designed to help contributors run all the checks on their contributions every time they commit.
+
+The following three style checks are performed:
+
+* **PEP8-compliant code**
+
+	The code style for the `jwst` repository generally conforms to
+	[PEP8](https://peps.python.org/pep-0008/), and the code style rules are enforced
+	using [Ruff](https://docs.astral.sh/ruff/). To run these checks standalone,
+	use the command
+
+        >> pre-commit run ruff
+
+    from within the `jwst` repository. Ruff will automatically pick up the appropriate configuration from the `.ruff.toml` and `pre-commit-config.yaml` files,
+	and perform only the checks that are turned on for our repository. To run ruff's
+	auto-formatter, which automatically fixes simple things like single vs double quotes, whitespace, etc., use the command
+
+	    >> pre-commit run ruff-format
+	
+	---
+	**Note:** If you run `ruff format .` from the top-level `jwst/` folder, it will re-format the entire code base.
+	Please do not do this; it makes pull requests more challenging to review. It's recommended to apply `ruff-format` through `pre-commit`.
+
+	---
+
+* **Numpy docstring style**
+
+	The docstring style for the `jwst` repository generally conforms to the
+	[Numpy style guide](https://numpydoc.readthedocs.io/en/latest/format.html), and the docstring
+	style rules are enforced using [numpydoc-validation](https://numpydoc.readthedocs.io/en/latest/validation.html).
+	To run these checks locally, use the command
+
+        >> pre-commit run numpydoc-validation
+
+
+* **PEP-compliant type hints**
+
+    The majority of the `jwst` repository does *not* have any type hints, and type hints are *not*
+    required for contributions. If type hints are used, though, their compliance with 
+	[PEP-484](https://peps.python.org/pep-0484/) standards
+    is enforced using [mypy](https://mypy.readthedocs.io/en/stable/index.html).
+    To run these checks locally, first `pip install mypy` then use the command
+
+        >> mypy .
```

Comment:
Does running this through pre-commit cause issues?

Reply:
Yeah I suppose we should just add mypy into the pre-commit and remove it from the other place in tox.ini, eh?

Reply:
Oh! I didn't release it wasn't already there. I vote for leaving that to a follow up PR since this one is so near completion. Maybe a follow-up issue/ticket instead.

Reply:
https://github.com/spacetelescope/jwst/issues/9103